django/db/backends/postgresql/schema.py

```diff
@@ -118,48 +126,31 @@ def _alter_column_type_sql(self, model, old_field, new_field, new_type):
             self.sql_alter_column_type += using_sql
         # Make ALTER TYPE with SERIAL make sense.
         table = strip_quotes(model._meta.db_table)
-        serial_fields_map = {
-            "bigserial": "bigint",
-            "serial": "integer",
-            "smallserial": "smallint",
+        auto_field_types = {
+            "AutoField",
+            "BigAutoField",
+            "SmallAutoField",
         }
-        if new_type.lower() in serial_fields_map:
+        old_is_auto = old_internal_type in auto_field_types
+        new_is_auto = new_internal_type in auto_field_types
+        if new_is_auto and not old_is_auto:
             column = strip_quotes(new_field.column)
             sequence_name = "%s_%s_seq" % (table, column)
             return (
                 (
                     self.sql_alter_column_type
                     % {
                         "column": self.quote_name(column),
-                        "type": serial_fields_map[new_type.lower()],
+                        "type": new_type,
                     },
                     [],
                 ),
                 [
                     (
-                        self.sql_delete_sequence
-                        % {
-                            "sequence": self.quote_name(sequence_name),
-                        },
-                        [],
-                    ),
-                    (
-                        self.sql_create_sequence
-                        % {
-                            "sequence": self.quote_name(sequence_name),
-                        },
-                        [],
-                    ),
-                    (
-                        self.sql_alter_column
+                        self.sql_add_identity
                         % {
                             "table": self.quote_name(table),
-                            "changes": self.sql_alter_column_default
-                            % {
-                                "column": self.quote_name(column),
-                                "default": "nextval('%s')"
-                                % self.quote_name(sequence_name),
-                            },
+                            "column": self.quote_name(strip_quotes(old_field.column)),
```

Comment:
Why we have here a `column` from the `old_field`?
```suggestion
                            "column": self.quote_name(column),
```